lib/authorizer/digest.js

```diff
@@ -90,6 +90,10 @@ function _getDigestAuthHeader (headers) {
 
 
 /**
+ * `realm` and `nonce` should be stored and reused for Digest auth. If not found or if the the server responds with 401,
+ * a second request should be sent using the values sent by the server.
```

Comment:
This should be `If not found, a second request....` 
Since we don't override the user provided values